modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1063,17 +1107,33 @@ public CompletableFuture<List<Table>> tablesAsync() {
      */
     private CompletableFuture<List<Table>> tablesAsyncInternal() {
         // TODO: IGNITE-16288 directTableIds should use async configuration API
-        return CompletableFuture.supplyAsync(this::directTableIds)
-                .thenCompose(tableIds -> {
-                    var tableFuts = new CompletableFuture[tableIds.size()];
+        return CompletableFuture.supplyAsync(() -> {
+            if (!busyLock.enterBusy()) {
+                throw new IgniteException(NODE_STOPPING_ERR, new NodeStoppingException());
+            }
+            try {
+                return directTableIds();
+            } finally {
+                busyLock.leaveBusy();
+            }
+        }).thenCompose(tableIds -> {
+            if (!busyLock.enterBusy()) {
+                throw new IgniteException(NODE_STOPPING_ERR, new NodeStoppingException());
+            }
+            try {
+                var tableFuts = new CompletableFuture[tableIds.size()];
```

Comment:
shouldn't we make some wrapper, like `withBusyLock(busyLock, lambda)` - maybe it will make the code more compact